pom.xml

```diff
@@ -137,10 +134,38 @@
 
     <profiles>
         <profile>
-            <id>release</id>
-            <properties>
-                <required.java.version>[1.8,1.9)</required.java.version>
-            </properties>
+            <id>toolchains</id>
+            <activation>
+                <property>
+                    <name>jdkToolchainVersion</name>
+                </property>
+            </activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-surefire-plugin</artifactId>
+                        <configuration>
+                            <jdkToolchain>
+                                <version>${jdkToolchainVersion}</version>
+                            </jdkToolchain>
+                        </configuration>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+
+        <profile>
```

Comment:
What are these profiles for? They don't seem to be needed.

Reply:
The only purpose is the following line:
 https://github.com/apache/mina-sshd/blob/38c19dc435a875e3cd98f17422f1441559632d28/sshd-ldap/pom.xml#L150
 
 LDAP testing requires [JVM specific args](https://github.com/apache/mina-sshd/blob/38c19dc435a875e3cd98f17422f1441559632d28/sshd-ldap/pom.xml#L36), but those are only available on JDK >= 11.  We can't use a simple profile activation, as those can only be activated based on the runtime JDK, not on the toolchain selected one.  Property based activation is also limited to presence or absence of a property. 

 But I think it should be possible to get rid of those by creating a property, let me simplify it.